docsource/paging.rst

```diff
@@ -0,0 +1,158 @@
+.. _paging:
+
+Paging workspace objects
+========================
+
+Workspaces can contain many objects, and the ``list_objects`` command returns a maximum of
+10000 ``object_info`` data structures (referred to as objects from now on) at a time. This
+section of the documentation discusses how to handle the case where access to large amounts
+of data is required.
+
+If it is reasonable to expect that a ``list_objects`` command will return no more than 10000
+objects then paging doesn't need to be considered. For example if the command only encompasses
+one workspace, only the most recent version of each object is required, and the ``max_objid``
+field in the workspace information returned by various API methods (for example
+``get_workspace_info``) is no more than 10000, then a single ``list_objects`` command can return
+all the data requested. See :ref:`workspaces` for more information about getting information
+about workspaces and :ref:`listobjects` for information about the ``list_objects`` command.
+
+There are currently two ways to page through workspace data - via object ID limits, and via
+reference limit.
+
+Parameters to avoid when paging
+-------------------------------
+
+Several parameters should be avoided when using the ``list_objects`` command to page
+through data:
+
+* Any of the time stamps (``after``, ``before``, ``after_epoch``, and ``before_epoch``)
+* ``savedby``
+* ``meta``
+
+The parameters are allowed, but strongly discouraged, when paging via object ID limits. Depending
+on the parameter set and the data to be listed, they may slow down the listing substantially,
+and may cause the data not to be sorted (see below).
+
+When paging via reference limit, these parameter are not allowed at all, and specifying a
+reference limit (the ``startfrom`` parameter) and one of the parameters above will cause an
+error. Additionally, the object ID limits (``minObjectID`` and ``maxObjectID``) are not allowed
+and including them will cause an error.
+
+Sort order
+----------
+
+When paging by reference limit and when paging via object ID limits without including any of
+the parameters to avoid listed above, data will be sorted by:
+
+1. the workspace ID, ascending
+2. the object ID, ascending
+3. the version, **descending**
+
+This makes it easy to determine how many versions an object has the first time it is encountered
+in the list of returned data.
+
+Object ID versus reference limits
+---------------------------------
+
+Paging by object ID limits works well for paging through data in a single workspace where only
+the most recent version of each object is required. Its advantage over reference based paging
```

Comment:
reference-based

(this is pretty pedantic -- I doubt most people would care)

Reply:
Fixed